src/lc0ctl/onnx2leela.cc

```diff
@@ -240,6 +317,11 @@ void ConvertOnnxToLeela() {
 
   const OptionsDict& dict = options_parser.GetOptionsDict();
 
+  auto onnx_model = ReadFileToString(dict.Get<std::string>(kInputFilenameId));
```

Comment:
Maybe isolate the code that modifies the ONNX from the code that embeds it into the Leela proto?

I can see two ways of doing this.
1. Either have a function that modifies ONNX first:
```cpp
std::string MaybeUpdateONNX(std::string input) {
   if (!fix50rule && !fixwdl) return input;
   pblczero::ModelProto model;
   model.ParseFromString(input);
   // Do the fixes.
   return model.OutputToString();
}

onnx->set_model(MaybeUpdateONNX(ReadFileToString(dict.Get<std::string>(kInputFilenameId))));
```

2. Or embed first and then postprocess:
```cpp
void MaybeUpdateWeights(pblczero::Net* out_weights) {
  if (!fix50rule && !fixwdl) return;
  // Otherwise parse from out_weights.mutable_onnx_model()->onnx(), fix and overwrite.
}

// And call only in the very end:
MaybeUpdateWeights(&out_weights);
WriteStringToGzFile(dict.Get<std::string>(kOutputFilenameId),
                      out_weights.OutputAsString());
```

Strictly speaking, fixing ONNX is not really part of the onnx to leela conversion (it may be a separate command or even tool), but I agree it's more convenient here. Do you think it would make sense also to have another command just for fixing the network? (maybe potentially not only ONNX).

Reply:
I originally structured the code to avoid multiple `ParseFromString()` calls, so I went with a variation of option 1 that allowed me to keep it like that.
 
Regarding a separate command to modify the onnx model, I don't think it makes sense as those are not general fixes but rather adjustments we do for differences between the tensorflow generated files and what our backend expects. So I don't see a use case a separate command. Even working around our limitations in handling models with external data requires two passes (either the current onnx2leela+leela2onnx or a hypothetical fix+onnx2leela flow), and even that should be fixed to work with just onnx2leela.